highlight/highlight
# Refactor current saved segment modals

## Summary
Refactor the saved segments code to lessen the amount of duplicate code when applying to logs and traces.

Makes the following updates:
1. Combine the UI of the saved segment modals into a shared component
2. Move feature specific modals into same directory
3. Remove the unnecessary attributes of a saved segment from the backend - ignored and not used ever
4. Fixes bug to allow user to duplicate and edit current saved segments

## How did you test this change?
1) Navigate to the sessions page
2) Add a filter so the default search is no longer used
- [ ] Can create a segment
- [ ] Can edit the name of a segment
- [ ] Can edit the search filters of a segment
- [ ] Can duplicate a segment
- [ ] Can delete a segment
3) Repeat steps 1-2 on the errors page
- [ ] Can create a segment
- [ ] Can edit the name of a segment
- [ ] Can edit the search filters of a segment
- [ ] Can duplicate a segment
- [ ] Can delete a segment

https://www.loom.com/share/ee344522b98f437fa198174777e214d0?sid=13fa4409-939b-46fb-b289-93ff9877aa00

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
N/A
